modules/lib/src/main/resources/assets/js/v6/features/shared/dialogs/delete/DeleteDialog.tsx

```diff
@@ -0,0 +1,166 @@
+                                            content={item}
+                                            showReferences
+                                            target={Branch.DRAFT}
+                                            hasInbound={isInbound(item.getContentId().toString())}
+                                            showOnlineStatus
+                                            mainItem
+                                            onClick={() => {
+                                                new EditContentEvent([item]).fire();
+                                            }}
+                                        />
+                                    ))}
+                                </ul>
+                            </div>
+
+                            <div className={cn('flex flex-col gap-y-7.5', dependants.length === 0 && 'hidden')}>
+                                <Separator className="" label={dependantsLabel} />
+                                <ul className="flex flex-col gap-y-2.5">
+                                    {dependants.map(item => (
+                                        <ContentItem
+                                            key={`dep-${item.getId()}`}
+                                            content={item}
+                                            showReferences
+                                            target={Branch.DRAFT}
+                                            hasInbound={isInbound(item.getContentId().toString())}
+                                            showOnlineStatus
+                                        />
+                                    ))}
+                                </ul>
+                            </div>
+
+                        </Dialog.Body>
+
+                        <Dialog.Footer className="flex items-center gap-1">
+                            <Button variant="text" size="lg" className="text-error font-normal" label={deleteButtonLabel} disabled={!ready} onClick={() => void handleDelete()} />
+                            <Button variant="solid" size="lg" className="font-normal"  label={archiveButtonLabel} disabled={!ready} onClick={() => void handleArchive()} />
+                        </Dialog.Footer>
+                    </Dialog.Content>
+                </Dialog.Portal>
+            </Dialog.Root>
+
+            <DialogPresetConfirmDelete
+                open={confirmOpen}
+                title={confirmTitle}
+                description={confirmDescription}
+                expected={total}
+                onConfirm={() => {
+                    setConfirmOpen(false);
+                    void executeDeleteDialogAction(confirmAction);
+                    setConfirmAction('delete');
+                }}
+                onCancel={() => {
+                    setConfirmOpen(false);
+                    setConfirmAction('delete');
+                }}
+            />
```

Comment:
You add DeleteDialog to the root of the app and always render it. But confirmation should not be always present in the DOM. Render it only when you require. Might not even need a separate Portal and overlay for it.